libraries/AP_SmartRTL/AP_SmartRTL.cpp

```diff
@@ -170,6 +170,25 @@ bool AP_SmartRTL::pop_point(Vector3f& point)
     return true;
 }
 
+// peek at next point on the path without removing it form the path. Returns true on success
+bool AP_SmartRTL::peek_point(Vector3f& point)
+{
+    // check we are active
+    if (!_active) {
+        return false;
+    }
+
+    // check we have another point
+    if (_path_points_count == 0) {
+        return false;
+    }
+
+    // return last point
+    // ToDo: do we need to use a semaphore?
```

Comment:
I've added the semaphore. This could lead to the vehicle stopping at the next point instead of flying through it but let's see.  I suspect it will be very rare.